packages/wrangler/src/index.tsx

```diff
@@ -1082,20 +1089,18 @@ function createCLIParser(argv: string[]) {
         }
 
         if (args["experimental-public"]) {
-          logger.warn(
-            "The --experimental-public field is experimental and will change in the future."
-          );
-        }
-
-        if (args["experimental-public"] && (args.site || config.site)) {
           throw new Error(
-            "Cannot use --experimental-public and a Site configuration together."
+            "The --experimental-public field has been renamed to --assets"
           );
         }
 
         if (args.public) {
+          throw new Error("The --public field has been renamed to --assets");
+        }
+
+        if ((args.assets || config.assets) && (args.site || config.site)) {
```

Comment:
I would split these out to make the error message more specific about what to fix:
```js
        if (args.assets && args.site) {
          throw new Error("Cannot use --assets and --site in the same Worker.");
        }
```

Reply:
It's actually a combination of each, so there are 4 version of this error we'd need ha. But fair, I'll do it.

Reply:
wait maybe I'll just push back on this? I'm happy to do it in a subsequent PR, or I'll end up bikeshedding the best way to do this here

Reply:
Yeah I guess its just a UX improvement, can be a follow-up